.github/workflows/python-poetry-ci.yml

```
@@ -0,0 +1,41 @@
+name: Linting
+on:
+  workflow_call:
+
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+
+    name: Style checks
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Poetry
+        uses: matrix-org/setup-python-poetry@v1
+
+      - name: Import order (isort)
+        run: poetry run isort --check --diff .
+
+      - name: Code style (black)
+        run: poetry run black --check .
+
+      - name: Semantic checks (flake8)
+        run: poetry run flake8
```

Not to bikeshed (don't block on this) but I've found GitHub renders steps quite reasonably when names are omitted... wonder if we'd be better / equally served by pervasively doing:

```suggestion
      - uses: actions/checkout@v2
      - uses: matrix-org/setup-python-poetry@v1
      - run: poetry run isort --check --diff .
      - run: poetry run black --check .
      - run: poetry run flake8
```